packages/propel/src/dialog/root.tsx

```
@@ -3,75 +3,114 @@
-        <div
-          className={cn(
-            "rounded-lg bg-custom-background-100 text-left shadow-custom-shadow-md transition-all w-full",
-            width,
-            className
-          )}
+// Constants
+const OVERLAY_CLASSNAME = cn("fixed inset-0 z-backdrop bg-custom-backdrop");
+const BASE_CLASSNAME = "relative text-left bg-custom-background-100 rounded-lg shadow-md w-full z-modal";
+
+// Utility functions
+const getPositionClassNames = React.useCallback((position: DialogPosition) => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)